mParticle-Apple-SDK/Kits/MPKitContainer.mm

```diff
@@ -294,10 +294,13 @@ - (void)initializeKits {
     
     for (NSDictionary *kitConfigurationDictionary in directoryContents) {
         MPKitConfiguration *kitConfiguration = [[MPKitConfiguration alloc] initWithDictionary:kitConfigurationDictionary];
-        self.kitConfigurations[kitConfiguration.integrationId] = kitConfiguration;
-        [self startKit:kitConfiguration.integrationId configuration:kitConfiguration];
-        
-        self.kitsInitialized = YES;
+        BOOL shouldStartKit = !(_disabledKits != nil && [_disabledKits containsObject:kitConfiguration.integrationId]);
+                if (shouldStartKit) {
+                    self.kitConfigurations[kitConfiguration.integrationId] = kitConfiguration;
+                    [self startKit:kitConfiguration.integrationId configuration:kitConfiguration];
+
+                    self.kitsInitialized = YES;
```

Comment:
Nit. `kitsInitialized` will be set to `YES` for every iteration of the loop. It might be better to just check the list of configured kits once the loop is done, then set it to `YES`, kind of like how  `self.sideloadedKits.count` is being checked below.

Reply:
i concur